packages/nx/src/migrations/update-16-8-0/escape-dollar-sign-env-variables.ts

```diff
@@ -0,0 +1,63 @@
+import { Tree } from '../../generators/tree';
+import { getProjects } from '../../generators/utils/project-configuration';
+
+/**
+ * This function escapes dollar sign in env variables
+ * It will go through:
+ * - '.env', '.local.env', '.env.local'
+ * - .env.[target-name], .[target-name].env
+ * - .env.[target-name].[configuration-name], .[target-name].[configuration-name].env
+ * - .env.[configuration-name], .[configuration-name].env
+ * at each project root and workspace root
+ * @param tree
+ */
+export default async function escapeDollarSignEnvVariables(tree: Tree) {
+  for (const [_, configuration] of getProjects(tree).entries()) {
+    const envFiles = ['.env', '.local.env', '.env.local'];
```

Comment:
These should be pushed once for the root.

Reply:
changed